### PR TITLE
Feature/syal 1517

### PR DESCRIPTION
**Pull request by HieuLM-AVISPL**
Hi @MaksimR-AVISPL , This is the code of Haivision Media Platform adapter. Please help me to review it. Thanks

### CODE REVIEW

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java [hunk 1]
```diff
@@ -0,0 +1,1200 @@
...
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		reentrantLock.lock();
+		try {
+			if (!checkValidCookieSession()) {
+				throw new FailedLoginException("Please enter valid password and username field.");
+			}
+			Map<String, String> statistics = new HashMap<>();
+			ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+			getContentValue();
+			retrieveDeviceInformationByPage(1);
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java:423]: Is it going to be enough to only have 1 page fetched here? Or is it just a placeholder for a future paginated fetch implementation?

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java [hunk 2]
```diff
@@ -0,0 +1,1200 @@
...
+			for (String command : commands) {
+				numPage = 1;
+				do {
+					nextPage = false;
+					response = doGetContentCommand(command, String.valueOf(numPage));
+					if (response != null && response.has(HaivisionMediaPlatformConstant.DATA)) {
+						for (JsonNode item : response.get(HaivisionMediaPlatformConstant.DATA)) {
+							switch (command) {
+								case HaivisionMediaPlatformCommand.SOURCE_COMMAND:
+									name = item.get(HaivisionMediaPlatformConstant.NAME).asText().trim();
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java]: can we guarantee these items will be not null, so .asText() won't trigger NPE?

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java [hunk 3]
```diff
@@ -0,0 +1,1200 @@
...
+
+	/**
+	 * Add advancedControllableProperties if advancedControllableProperties different empty
+	 *
+	 * @param advancedControllableProperties advancedControllableProperties is the list that store all controllable properties
+	 * @param stats store all statistics
+	 * @param property the property is item advancedControllableProperties
+	 * @throws IllegalStateException when exception occur
+	 */
+	private void addAdvanceControlProperties(List<AdvancedControllableProperty> advancedControllableProperties, Map<String, String> stats, AdvancedControllableProperty property, String value) {
```
Comment (MaksimR-AVISPL) [inline src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java]: I believe it was supposed to be Advance**d** properties?  :)

**Review by MaksimR-AVISPL — APPROVED** [review]

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java (hunk 1 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java:423]: > Is it going to be enough to only have 1 page fetched here? Or is it just a placeholder for a future paginated fetch implementation?

I only take 1 page so I can know the total number of pages and the total number of devices. After  I will retrieve all page.
image

File: src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java (hunk 2 above)
Comment (HieuLM-AVISPL) [reply src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java]: > can we guarantee these items will be not null, so .asText() won't trigger NPE?

According to the documentation, these items always exist, but I will handle them at risk later

**Pull request merged**